### ISSUE TITLE
[Spreadsheet template] Review structure and decide whether to omit fields or enforce 1:1 relationships

### ISSUE THREAD

**Issue by duncandewhurst** [GFDRR/rdls-spreadsheet-template#1]
@stufraser1 @matamadio I've generated a [basic spreadsheet template](https://docs.google.com/spreadsheets/d/1eXlt20hw-1KVXnND1Kj9VZIracai-VwXFwBpJK7ijeU/edit#gid=1535404388) using Flatten Tool so that you can get a sense of the scale and structure of a template which contains all the fields in the JSON schema and which permits one-to-many relationships for all arrays.

Please review it and let me know your thoughts on whether you want to omit any fields or enforce 1:1 relationships for any arrays.

This spreadsheet represents the basic Flatten Tool output and doesn't include any of the other features discussed in the scoping document (field metadata; data validation; formatting; worksheet grouping; ordering and colouring etc.)

**Comment by matamadio** [comment]
Thanks, I'll look through and share comments.

**Comment by matamadio** [comment]
One question, if it makes sense: according to #70 and [#95](https://github.com/GFDRR/rdl-standard/issues/95), isn't `hazard_event_sets` supposed to be just one per dataset, including n (array) events?
Than could it be merged at the same level of the dataset tab?
Sorry if I'm missing something.

Example (transposed):

```
identifier
title
description
risk_data_type
publisher/name
publisher/email
publisher/url
version
purpose
project
details
spatial/countries
spatial/bbox
spatial/geometry
spatial/centroid
spatial/scale
license
contact_point/name
contact_point/email
contact_point/url
creator/name
creator/email
creator/url
temporal_resolution

hazard/event_sets/0/id
hazard/event_sets/0/analysis_type
hazard/event_sets/0/frequency_distribution
hazard/event_sets/0/seasonality
hazard/event_sets/0/calculation_method
hazard/event_sets/0/event_count
hazard/event_sets/0/occurrence_range
hazard/event_sets/0/spatial/countries
hazard/event_sets/0/spatial/bbox
hazard/event_sets/0/spatial/geometry
hazard/event_sets/0/spatial/centroid
hazard/event_sets/0/spatial/scale
hazard/event_sets/0/temporal/start
hazard/event_sets/0/temporal/end
hazard/event_sets/0/temporal/duration

exposure/category
exposure/taxonomy
vulnerability/hazard_primary
vulnerability/hazard_secondary
vulnerability/hazard_process_primary
vulnerability/hazard_process_secondary
vulnerability/hazard_analysis_type
vulnerability/intensity
vulnerability/category
vulnerability/taxonomy
vulnerability/impact/type
vulnerability/impact/metric
vulnerability/impact/unit
vulnerability/impact/base_data_type
vulnerability/spatial/countries
vulnerability/spatial/bbox
vulnerability/spatial/geometry
vulnerability/spatial/centroid
vulnerability/spatial/scale
vulnerability/functions/vulnerability/approach
vulnerability/functions/vulnerability/relationship
vulnerability/functions/fragility/approach
vulnerability/functions/fragility/relationship
vulnerability/functions/fragility/damage_scale_name
vulnerability/functions/fragility/damage_states_names
vulnerability/functions/damage_to_loss/approach
vulnerability/functions/damage_to_loss/relationship
vulnerability/functions/damage_to_loss/damage_scale_name
vulnerability/functions/damage_to_loss/damage_states_names
vulnerability/functions/engineering_demand/parameter
vulnerability/functions/engineering_demand/approach
vulnerability/functions/engineering_demand/relationship
vulnerability/analysis_details
vulnerability/se_category/scheme
vulnerability/se_category/id
vulnerability/se_category/description
vulnerability/se_category/uri
loss/hazard_type
loss/hazard_process
loss/description
loss/category
loss/impact/type
loss/impact/metric
loss/impact/unit
loss/impact/base_data_type
loss/type
loss/approach
loss/hazard_analysis_type
loss/hazard_id
loss/exposure_id
loss/vulnerability_id
```

**Comment by stufraser1** [comment]
- On 'datasets', could 'temporal' sit with spatial and publisher, creator all sit together to group similar fields?
- What does the '0' refer to in the field names? 
- Requires some reorganisation of hazards sheets might be helpful, so its more ordered as the hierarchy you would complete it. I.e., hazard_event_sets_hazards before _events before _footprints.
- In fact, reorganisation of all sheets would be helpful - alphabetical order of sheets is maybe less helpful than ordering by component in the logical order someone would complete the metadata.

**Comment by duncandewhurst** [comment]
> Looks good. I don't think it's worth to strip out any tab; if some are not used, could those be removed by the user?

Yes, users can remove sheets. We can include instructions to that effect in the documentation.

> * On 'datasets', could 'temporal' sit with spatial and publisher, creator all sit together to group similar fields?

Yes. However, the fields are ordered according to their order in the RDLS schema so we should fix the ordering there. That way, anything that we generate from the schema (such as the schema browser and reference tables in the RDLS documentation) will also have the desired ordering. I'll open an issue on the main RDLS repo.

>     * What does the '0' refer to in the field names?

It indicates that each row under that field path should be interpreted as an item in an array, e.g. under `attributions/0/id` the first row will be interpreted as the `id` of the first item in the `attributions` array and the second row will be interpreted as the `id` of the second item, etc. We can explain this in the documentation.

>     * Requires some reorganisation of hazards sheets might be helpful, so its more ordered as the hierarchy you would complete it. I.e., hazard_event_sets_hazards before _events before _footprints.
>
>     * In fact, reorganisation of all sheets would be helpful - alphabetical order of sheets is maybe less helpful than ordering by component in the logical order someone would complete the metadata.

Sounds good. It should be possible to order the sheets based on the order of the fields in the schema so we can tackle this at the same time as ordering the fields in the schema.

**Comment by matamadio** [comment]
Please check the enums in the template, they seem to bring the wrong columns.
![immagine](https://github.com/GFDRR/rdls-spreadsheet-template/assets/44863827/1fa9809d-5d59-4215-94dc-86274af66473)

**Comment by duncandewhurst** [comment]
Ah yes, the codelist validation in everything but the datasets worksheet was out of sync. I've fixed that in the updated version shared in #3

**Comment by duncandewhurst** [comment]
Closing since the sheets were reordered, field order will be addressed as part of https://github.com/GFDRR/rdl-standard/issues/177, and the readme contains documentation on hiding unneeded sheets.
